jraft-core/src/main/java/com/alipay/sofa/jraft/storage/snapshot/SnapshotExecutorImpl.java

```diff
@@ -324,6 +324,17 @@ public void doSnapshot(final Closure done) {
                 Utils.runClosureInThread(done);
                 return;
             }
+
+            if (this.fsmCaller.getLastAppliedIndex() - this.lastSnapshotIndex < this.node.getOptions()
+                .getSnapshotLogIndexMargin()) {
+                // If state machine's lastAppliedIndex value minus lastSnapshotId value is
+                // less than or equal snapshotIntervalDist value, then directly return.
+                doUnlock = false;
```

Comment:
@zongtanghu  Can you add a debug log here?  Log the margin and says that ignore this snapshot time by snapshotLogIndexMargin setting.

Reply:
Okay, I will add the debug log that you said.